internal/tfprotov5tov6/tfprotov5tov6_test.go

```diff
@@ -2990,6 +3013,131 @@ func TestValidateListResourceConfigResponse(t *testing.T) {
 	}
 }
 
+func TestListResourceRequest(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		in       *tfprotov5.ListResourceRequest
+		expected *tfprotov6.ListResourceRequest
+	}{
+		"nil": {
+			in:       nil,
+			expected: nil,
+		},
+		"all-valid-fields": {
+			in: &tfprotov5.ListResourceRequest{
+				Config:   &testTfprotov5DynamicValue,
+				TypeName: "test_list_resource",
+			},
+			expected: &tfprotov6.ListResourceRequest{
+				Config:   &testTfprotov6DynamicValue,
+				TypeName: "test_list_resource",
+			},
+		},
+	}
+
+	for name, testCase := range testCases {
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := tfprotov5tov6.ListResourceRequest(testCase.in)
+
+			if diff := cmp.Diff(got, testCase.expected); diff != "" {
+				t.Errorf("unexpected difference: %s", diff)
+			}
+		})
+	}
+}
+
+func TestListResourceServerStream(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		in       *tfprotov5.ListResourceServerStream
+		expected *tfprotov6.ListResourceServerStream
+	}{
+		"nil": {
+			in:       nil,
+			expected: nil,
+		},
+		"all-valid-fields": {
+			in:       testStreamv5,
+			expected: testStreamv6,
+		},
+	}
+
+	for name, testCase := range testCases {
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := tfprotov5tov6.ListResourceServerStream(testCase.in)
+
+			if got == nil {
+				if diff := cmp.Diff(got, testCase.expected); diff != "" {
+					t.Errorf("unexpected difference: %s", diff)
+				}
+			} else {
+				resultSlice := make([]tfprotov6.ListResourceResult, 0)
```

Comment:
Suggestion: we can use [`slices.Collect`](https://pkg.go.dev/slices#Collect) – the reverse of `slices.Values`.